### PR TITLE
test: implement frontend e2e testing using playwright

### PR DESCRIPTION
**Pull request by st3iny**
## TODO

- [x] Actually serve a nextcloud server via https so that playwright can access it

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -138,6 +138,90 @@ jobs:
...
+          extensions: ctype,curl,dom,gd,iconv,intl,json,mbstring,openssl,posix,sqlite,xml,zip,gmp,oci8
+          coverage: xdebug
+      - name: Checkout Nextcloud
+        run: git clone https://github.com/nextcloud/server.git --recursive --depth 1 -b ${{ matrix.nextcloud-version }} nextcloud
+      - name: Check out the app
+        uses: actions/checkout@v3
+        with:
+          path: nextcloud/apps/twofactor_webauthn
+      - name: Install Nextcloud
+        run: php -f nextcloud/occ maintenance:install --database mysql --database-host 127.0.0.1 --database-name nextcloud --database-user nextcloud --database-pass nextcloud --admin-user admin --admin-pass admin
```
Comment (ChristophWurst) [inline .github/workflows/test.yml:182]: https://github.com/ChristophWurst/setup-nextcloud :wink:

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -138,6 +138,90 @@ jobs:
...
       env:
           CI: true
+
+  frontend-e2e-tests:
+    runs-on: ubuntu-latest
+    name: Front-end E2E tests
+    strategy:
+      matrix:
+        php-version: [ '8.1' ]
+        nextcloud-version: [ 'master' ]
```
Comment (ChristophWurst) [inline .github/workflows/test.yml:151]: would it make sense to test stable* too?

**Review by ChristophWurst — APPROVED** [review]
:sunglasses:

File: .github/workflows/test.yml (hunk 1 above)
Comment (st3iny) [reply .github/workflows/test.yml:182]: Well :see_no_evil: I forgot about that one.

I'll refactor the action.

File: .github/workflows/test.yml (hunk 2 above)
Comment (st3iny) [reply .github/workflows/test.yml]: I added the oldest and most recent supported versions. As this repo is not seeing a lot of activity and there are only 2 test cases it should be fine.

File: .github/workflows/test.yml (hunk 1 above)
Comment (st3iny) [reply .github/workflows/test.yml]: Done

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**